src/pass_manager.cpp

```diff
@@ -41,6 +41,12 @@ inline namespace MIGRAPHX_INLINE_NS {
 MIGRAPHX_DECLARE_ENV_VAR(MIGRAPHX_TRACE_PASSES);
 MIGRAPHX_DECLARE_ENV_VAR(MIGRAPHX_TIME_PASSES);
 
+template <class T>
+std::vector<T*> generic_get_modules(T* mm);
+
+template <>
+std::vector<module*> generic_get_modules(module* mm);
```

Comment:
Yea I dont think we should put specialization for these template functions. Actually, its much shorter to just push the current module to the front:

```cpp
auto mods = root_mod->get_submodules();
mods.insert(mods.begin(), root_mod);
```

Reply:
Done